### PR TITLE
feat: Provide React and Preact bundles.

### PR DESCRIPTION
**Pull request by rokbar-nosto**
## Checklist

- [ ] Pull Request is properly described.
- [ ] The corresponding Jira ticket is updated.
- [ ] I have requested a review from at least one reviewer.
- [ ] The changes are tested
- [ ] I have verified my UX changes with a designer
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: src/defaults/Autocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                                <div className="ns-autocomplete-product-brand">
+                                    {hit.brand}
+                                </div>
+                            )}
+                            <div className="ns-autocomplete-product-name">
+                                {hit.name}
+                            </div>
+                            <div>
+                                <span>{hit.price}&euro;</span>
+                                {hit.listPrice && (
```
Comment (timowestnosto) [inline src/defaults/Autocomplete.tsx:147]: additional condition: hit.price !== hit.listPrice

File: src/defaults/Autocomplete.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    const hasKeywords = !!response?.keywords?.hits?.length
+    const hasProducts = !!response?.products?.hits?.length
+    const hasHistory = !!history?.length
+
+    if (!hasKeywords && !hasProducts && !hasHistory) {
+        return null
+    }
+
+    return (
+        <div className="ns-autocomplete-results">
```
Comment (timowestnosto) [inline src/defaults/Autocomplete.tsx:27]: should we parametrize the class names in these components?

Comment (rokbar-nosto) [reply src/defaults/Autocomplete.tsx:27]: Good idea. Same as we want to do in search templates.

File: scripts/exportTemplates.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import * as fs from 'fs'
+
+// write script that generates const variable string from /src/defaults/autocomplete.liquid
+function generateAutocompleteLiquid() {
+    const autocompleteLiquid = fs.readFileSync('./src/defaults/autocomplete.liquid', 'utf8')
+    const autocompleteLiquidLines = autocompleteLiquid.split('\n')
+    const autocompleteLiquidLinesWithConst = `export const defaultLiquidTemplate = \`\n${autocompleteLiquidLines.map(line => `${line}`).join('\n')}\n\`\n`
```
Comment (timowestnosto) [inline scripts/exportTemplates.js:7]: can you use autocompleteLiquid directly here?

File: scripts/exportTemplates.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    fs.appendFileSync('./src/defaults/_generated.ts', autocompleteLiquidLinesWithConst)
+}
+
+function generateAutocompleteMustache() {
+    const autocompleteMustache = fs.readFileSync('./src/defaults/autocomplete.mustache', 'utf8')
+    const autocompleteMustacheLines = autocompleteMustache.split('\n')
+    const autocompleteMustacheLinesWithConst = `export const defaultMustacheTemplate = \`\n${autocompleteMustacheLines.map(line => `${line}`).join('\n')}\n\`\n`
```
Comment (timowestnosto) [inline scripts/exportTemplates.js:15]: can you use autocompleteMustache directly here?

File: scripts/exportTemplates.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+    fs.appendFileSync('./src/defaults/_generated.ts', autocompleteMustacheLinesWithConst)
+}
+
+(() => {
```
Comment (timowestnosto) [inline scripts/exportTemplates.js:20]: is the IIFE wrapping needed?

File: scripts/postCommit.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { exec } from "child_process"
+
+const generatedFile = "src/defaults/_generated.ts"
+
+;(() => {
```
Comment (timowestnosto) [inline scripts/postCommit.js:5]: is the IIFE wrapping needed?

File: src/config.ts [hunk 7]
```diff
@@ -33,7 +33,7 @@ export interface AutocompleteConfig<State> {
     /**
      * The function to use to submit the search
      */
-    submit: (query: string, config?: AutocompleteConfig<State>) => unknown
+    submit?: <T extends State>(query: string, config: AutocompleteConfig<T>) => unknown
```
Comment (timowestnosto) [inline src/config.ts:36]: the generic parameter doesn't seem to be used
drop it?

File: src/config.ts [hunk 8]
```diff
@@ -69,15 +69,20 @@ export const defaultConfig = {
...
-        search(
-            {
-                query,
-            },
-            {
-                redirect: true,
-                track: config.nostoAnalytics ? "serp" : undefined,
-            }
-        )
+    submit: <State>(query: string, config: AutocompleteConfig<State>) => {
```
Comment (timowestnosto) [inline src/config.ts:72]: State as generic parameter?
above (line 36) it is used as a type, which is confusing

**Comment by timowestnosto** [comment]
Do we need the pre & post commit hooks? Can't we add the extra artifacts in the build phase?

File: scripts/exportTemplates.js (hunk 5 above)
Comment (rokbar-nosto) [reply scripts/exportTemplates.js:20]: Indeed no, calling function is enough.

File: src/utils/ga.ts [hunk 9]
```diff
@@ -86,9 +86,9 @@ export const getGaTrackUrl = <State>(
 ) => {
     const gaConfig = isGaEnabled(config)
         ? typeof config.googleAnalytics === "boolean"
-            ? defaultConfig.googleAnalytics
+            ? getDefaultConfig().googleAnalytics
```
Comment (timowestnosto) [inline src/utils/ga.ts:89]: getDefaultConfig() should only be used to merge defaults into the full configuration

File: src/utils/ga.ts [hunk 10]
```diff
@@ -86,9 +86,9 @@ export const getGaTrackUrl = <State>(
...
             : {
-                  ...defaultConfig.googleAnalytics,
+                  ...getDefaultConfig().googleAnalytics,
```
Comment (timowestnosto) [inline src/utils/ga.ts:91]: same as above

File: scripts/exportTemplates.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
+import * as fs from "fs"
+
+// write script that generates const variable string from /src/defaults/autocomplete.liquid
+function generateAutocompleteLiquid() {
```
Comment (timowestnosto) [inline scripts/exportTemplates.js:4]: combine into single function and just overwrite the file?

**Review by timowestnosto — APPROVED** [review]

**Pull request merged**